internal/osbuild2/rpm_stage.go

```diff
@@ -33,15 +33,23 @@ type RPMPackage struct {
 
 func (RPMStageOptions) isStageOptions() {}
 
+// RPMStageInputs defines a collection of packages to be installed by the RPM
+// stage.
 type RPMStageInputs struct {
+
+	// Packages to install
 	Packages *RPMStageInput `json:"packages"`
 }
 
 func (RPMStageInputs) isStageInputs() {}
 
+// RPMStageInput defines a single input source.
 type RPMStageInput struct {
 	inputCommon
-	References RPMStageReferences `json:"references"`
+
+	// Collection of references. Each reference defines a package to be
+	// installed, with optional metadata.
+	References RPMStageSourceArrayRefs `json:"references"`
```

Comment:
Nitpick: You could probably use the `References` interface as a type, but it may actually complicate code a bit. This would allow the structure to be agnostic of the specific Reference implementation.

Reply:
Fair enough as well... 😉